Hunting Queries/AWSCloudTrail/AWS_PrivilegedRoleAttachedToInstance.yaml

```diff
@@ -0,0 +1,34 @@
+id: 0db42a94-e7c8-4bf1-99a7-1a2fb4158212
+name: Privileged role attached to Instance
+description: |
+  Identity and Access Management (IAM) securely manages access to AWS services and resources. 
+  This query looks for when Privileged role is attached to existing instance or new instance at deployment. This instance may be used by adversary to escalate normal user privileges to adminsitrative user.
+requiredDataConnectors:
+  - connectorId: AWS
+    dataTypes:
+      - AWSCloudTrail
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1078
+query: |
+
+  let EventNameList = dynamic(["AttachUserPolicy","AttachRolePolicy"]);
+  let PolicyArnList = dynamic(["arn:aws:iam::aws:policy/AdministratorAccess","arn:aws:iam::aws:policy/DatabaseAdministrator","arn:aws:iam::aws:policy/NetworkAdministrator","arn:aws:iam::aws:policy/SystemAdministrator","arn:aws:iam::aws:policy/AmazonS3FullAccess"]);
+  let timeframe = 1d;
+  let lookback = 14d;
+  let PrivilegedRoleorUsers = AWSCloudTrail
+  | where TimeGenerated >= ago(lookback) 
+  | where EventName in (EventNameList)
+  | extend PolicyArn = tostring(parse_json(RequestParameters).policyArn), RoleName = tostring(parse_json(RequestParameters).roleName)
+  | where PolicyArn in (PolicyArnList)
+  | distinct PolicyArn, UserIdentityType, UserIdentityUserName,RoleName;
+  PrivilegedRoleorUsers
+  | join (
+  AWSCloudTrail
+  | where TimeGenerated >= ago(timeframe)
+  | where EventName in ("AddRoleToInstanceProfile") 
+  | extend InstanceProfileName = tostring(parse_json(RequestParameters).InstanceProfileName), RoleName = tostring(parse_json(RequestParameters).roleName)
+  | summarize count(), min(TimeGenerated), max(TimeGenerated) by EventSource, EventName, UserIdentityType , UserIdentityArn , UserIdentityUserName, SourceIpAddress, RoleName
```

Comment:
It is best to label these, such as | summarize EventCount = count(), StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated)